lib/elixir/lib/module.ex

```diff
@@ -104,6 +104,17 @@ defmodule Module do
   Multiple uses of `@compile` will accumulate instead of overriding
   previous ones. See the "Compile options" section below.
 
+  ### `@deprecated`
+
+  Provides the deprecation reason for a function. For example:
+
+      defmodule MyModule do
+        @deprecated "Use MyModule.hello_world/0 instead"
+        def hello, do: "world"
+
+        def hello_world, do: "Hello world!"
```

Comment:
Could we use a real example like any function that has been deprecated in elixir?

Reply:
Should I already add the @deprecated to the functions deprecated on `elixir_disptach.erl`

Reply:
> Should I already add the @deprecated to the functions deprecated on elixir_disptach.erl?

No. :) Those are a bit trickier because this proposal will only warn for Mix projects. And we want the Elixir ones to always warn (in scripts, etc). We still need to find a way to solve it.